reports-backend/src/main/java/org/jboss/da/listings/impl/service/WhiteArtifactServiceImpl.java

```diff
@@ -48,6 +51,41 @@
     }
 
     @Override
+    public List<ProductVersion> toProductsContainingOnlyWhiteArtifacts(List<ProductVersion> products){
+    	
+    	// Get all version artifacts
+    	List<ProductVersion> productVersions = new ArrayList<>(products);
```

Comment:
Why do you copy the list? You don't modify the list but the ProductVersions inside.